### PR TITLE
Modules: Group properties

### PR DESCRIPTION
**Pull request by ifarzana**
- Fixes https://github.com/xibosignageltd/xibo-private/issues/175

### CODE REVIEW

File: lib/Widget/Definition/GroupProperty.php
Comment (dasgarner) [inline lib/Widget/Definition/GroupProperty.php:1]: Would it make more sense for this to be called `PropertyGroup` to match the XML?

File: lib/Entity/Module.php [hunk 1]
```diff
@@ -221,6 +221,16 @@ class Module implements \JsonSerializable
...
 
+    /**
+     * @SWG\Property(
+     *     description="An array of additional module specific group properties",
+     *     type="array",
+     *     @SWG\Items(type="string")
+     * )
+     * @var \Xibo\Widget\Definition\GroupProperty[]
+     */
+    public $propertyGroups = [];
```
Comment (dasgarner) [inline lib/Entity/Module.php:232]: @maurofmferrao are we insisting that the property groups are defined at the module level? Or will we have those also on the Module Template and combine that on the front end when rendering the form?

**Review by dasgarner — CHANGES_REQUESTED** [review]

File: lib/Entity/Module.php (hunk 1 above)
Comment (maurofmferrao) [reply lib/Entity/Module.php:232]: Hmm yeah, I think it would be nice to have them all in the templates as well 🤔

Comment (dasgarner) [reply lib/Entity/Module.php:232]: I think it comes down to the question of whether as a template developer, would you want to add your own tabs, or be limited to the ones provided by the module developer?

Comment (maurofmferrao) [reply lib/Entity/Module.php:232]: That a good question. I see the possibility of a template having too many properties and have to group them. Something like Metro Static template can use different groups than other "Twitter" widgets, So maybe we should also had that property groups for the template as well

**Review by dasgarner — APPROVED** [review]

**Pull request merged**